src/plugin/systems/rigid_body.rs

```diff
@@ -418,6 +710,88 @@ pub fn writeback_rigid_bodies(
                     }
                 }
             }
+
+            (my_new_global_transform, delta_transform, my_velocity)
+        } else {
+            (parent_global_transform, parent_delta, parent_velocity)
+        };
+
+        recurse_child_transforms(
+            context,
+            config,
+            sim_to_render_time,
+            writeback,
+            transform,
+            delta_transform,
+            velocity,
+            children_query,
+            child,
+            world_offset,
+        );
+    }
+}
+
+/// Syncs up child velocities with their parents in the physics simulation.
+/// This is done to avoid child components getting hit by their parent and rapier
+/// assuming the child is hit by the full velocity of the parent instead of `parent vel - child vel`.
+///
+/// This will not change the bevy component's velocity.
+pub fn sync_vel(
```

Comment:
I assume this is to address https://github.com/dimforge/bevy_rapier/issues/383

Reply:
This could be stripped out of this PR for sure, it is not at all necessary for the worlds changes.